src/frontend/views/games_view.py

```diff
@@ -39,10 +43,10 @@ def get_queryset(self):
     self.network = get_object_or_404(Network, name=self.kwargs["network"])
 
     if self.kwargs["kind"] == "training":
-      games = TrainingGame.objects.filter(white_network=self.network).order_by("-created_at")
+      games = TrainingGame.objects.filter(white_network=self.network).order_by("-created_at").prefetch_related("submitted_by")
```

Comment:
Nice to know.

Do you know the performance impact of just this ?

I think this is the real pain / culprit and is alone responsible of 99% of the slowness (much more that time format or reverse.

Reply:
It cut down the load time of some of the pages by about 70%-90%. Yay! :)